src/Microsoft.AspNet.Mvc.ModelBinding/Binders/FormFileModelBinder.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    public class FormFileModelBinder : IModelBinder
+    {
+        public async Task<bool> BindModelAsync([NotNull] ModelBindingContext bindingContext)
+        {
+            var request = bindingContext.OperationBindingContext.HttpContext.Request;
+            if (request.HasFormContentType)
+            {
+                var form =  await request.ReadFormAsync();
+                var postedFiles = new List<IFormFile>();
+
+                foreach (var file in form.Files)
+                {
+                    if (string.IsNullOrEmpty(file.ContentDisposition))
+                    {
+                        continue;
+                    }
+
+                    var parsedContentDisposition = file.ParseContentDisposition();
+
+                    var modelName = parsedContentDisposition.Key;
+
+                    if (modelName.Equals(bindingContext.ModelName))
+                    {
+                        postedFiles.Add(file);
+                    }
+                }
+
+                if (!postedFiles.Any())
+                {
+                    return false;
+                }
+                if (bindingContext.ModelType == typeof(IFormFile))
```

Comment:
These kind of checks should be in the beginning of this binder too as we do not want to do any kind of processing if the model type is entirely different